src/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\Profile;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Hash;
+use App\Http\Requests\RegisterRequest;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Auth\Events\Verified;
+
+class AuthController extends Controller
+{
+
+    public function register(RegisterRequest $request)
+    {
+        try {
+            $user = User::create([
+                'username' => $request->username,
+                'password' => Hash::make($request->password),
+                'email' => $request->email,
+                'status' => 0,
+                'remember_token' => Str::random(40),
+            ]);
+
+            Profile::create([
+                'user_id' => $user->id,
+                'first_name' => $request->first_name,
+                'last_name' => $request->last_name,
+                'middle_name' => $request->middle_name,
+                'birth_date' => $request->birthday,
+                'address' => implode(', ', array_filter([
+                    $request->lot_block,
+                    $request->street,
+                    $request->city,
+                    $request->province,
+                    $request->country,
+                    $request->zip_code,
+                ])),
+            ]);
+
+            auth()->login($user);
+            event(new Registered($user));
+
+            return redirect('/verify-waiting')->with('success', 'Registration is successful! Please check your email for verification.');
+        } catch (\Exception $e) {
+            dd($e->getMessage());
```

Comment:
handle the error appropriately, maybe displaying the error on the frontend not just using `dd`

Reply:
Replaced dd by displaying the error on frontend. Also added error bag in views.